### PR TITLE
fix: Update show maps to use an initiator view return object instead …

### PR DESCRIPTION
**Pull request by David-T-White**
…of host view

This commit resolves an issue in which chooseLun will always return 1 as the array returns initiator-view objects instead of host-view objects.

### CODE REVIEW

File: pkg/api/volumes.go [hunk 1]
```diff
@@ -366,6 +367,8 @@ func (client *Client) mapVolumeProcess(volumeName, initiatorName string, lun int
 		}
 	} else if metadata.ReturnCode == common.VolumeNotFoundErrorCode {
 		return status.Errorf(codes.NotFound, "volume %s not found", volumeName)
+	} else if metadata.ReturnCode == common.LUNOverlapErrorCode {
+		return status.Errorf(codes.AlreadyExists, "lun overlap for lun: %d", lun)
```
Comment (jskazinski) [inline pkg/api/volumes.go:371]: One thought it to make sure that CSI Sanity still passes when codes.AlreadyExists is returned. Many csi sanity test cases rely on very specific return codes.

File: pkg/regression/v2_volumes.go [hunk 2]
```diff
@@ -85,13 +86,43 @@ var _ = DescribeRegression("Volume Testing (v2)", func(tc *TestContext) {
...
+			Expect(err).To(BeNil())
+			Expect(status.ResponseTypeNumeric).To(Equal(storageapi.ApiSuccess))
+			ShowVolumes(logger, response)
+		})
+
+		It("should be able to publish/map a second volume", func() {
+
+			logger := klog.FromContext(tc.Config.Ctx)
+			lun, err := client.PublishVolume(volname2, tc.Config.StorageController.Initiator)
+			logger.V(3).Info("PublishVolume", "name", volname1, "lun", lun)
```
Comment (jskazinski) [inline pkg/regression/v2_volumes.go:110]: Should this be logger.V(3).Info("PublishVolume", "name", volname2, "lun", lun)

File: pkg/regression/v2_volumes.go [hunk 3]
```diff
@@ -85,13 +86,43 @@ var _ = DescribeRegression("Volume Testing (v2)", func(tc *TestContext) {
...
+			logger := klog.FromContext(tc.Config.Ctx)
+			volume, status, err := client.CreateVolume(volname2, size, tc.Config.StorageController.Pool, poolType)
+			logger.V(3).Info("CreateVolume", "name", volname2, "size", size, "wwn", volume.Wwn, "mc-response", status.ResponseTypeNumeric)
+
+			Expect(err).To(BeNil())
+			Expect(status.ResponseTypeNumeric).To(Equal(storageapi.ApiSuccess))
+			Expect(volume.Wwn).ShouldNot(BeEmpty())
+
+			// Dump volumes
+			response, status, err := client.ShowVolumes(volname1)
```
Comment (jskazinski) [inline pkg/regression/v2_volumes.go:100]: Should this be: response, status, err := client.ShowVolumes(volname2)

File: pkg/api/volumes.go (hunk 1 above)
Comment (David-T-White) [reply pkg/api/volumes.go:371]: I tested and confirmed this doesn't cause new csi sanity failures

File: pkg/regression/v2_volumes.go (hunk 2 above)
Comment (David-T-White) [reply pkg/regression/v2_volumes.go]: Fixed in commit 4a29dad

File: pkg/regression/v2_volumes.go (hunk 3 above)
Comment (David-T-White) [reply pkg/regression/v2_volumes.go]: Fixed in commit 4a29dad

File: pkg/api/volumes.go (hunk 1 above)
Comment (jskazinski) [reply pkg/api/volumes.go:371]: Thank you.

**Review by jskazinski — APPROVED** [review]
Looks good to me, thank you.

**Review by seagate-chris — APPROVED** [review]
Looks good.

**Pull request merged**